### ISSUE TITLE
🧟📄 v1.87.0 deletes of files not honored, zombies files come back from the dead 

### ISSUE THREAD

**Issue by clach04** [GitJournal/GitJournal#962]
Steps:

1. from a checkout (that GitJournal is not using)
2. Delete  file
3. commit the delete
4. push to sevver
5. in git journal refresh/pull
6. in other checkout pull - gi tlog shows GJ re-added the file :crying_cat_face: 

Older version with c-git lib did not have this issue.

Deleting file inside GitJournal works (but a pain if want to delete a bunch of files/notes.

Deleting from git dir on phone, then syncing does NOT work.

**Comment by davidluhr** [comment]
Having this issue as well. I had to uninstall the app because this was causing major issues for me. I spent a long time pruning and reorgranizing an extensive collection of notes, and GitJournal's sync re-added the notes in their original location. I had to hard reset the GitJournal commit and remove the app entirely to avoid this risk.

I hope there is a simple fix for this. Maybe some Git configuration changed?

**Comment by ptvirgo** [comment]
Probably goes without saying but if someone finds a workaround, please let us know.

**Comment by tschager** [comment]
The only workaround I know is to delete the app before the delete operations and re-install afterwards..

**Comment by clach04** [comment]
> The only workaround I know is to delete the app before the delete operations and re-install afterwards..

Untested but I'm guessing breaking the link to the git repo maybe another similar workaround without having to uninstall and then reinstall.

**Comment by clach04** [comment]
> I've unfortunately had to uninstall the app to avoid the issue. I miss it; it was a valuable part of my personal knowledge management and without it I don't have a way to edit my notes without a computer.
> 
> I'm hoping there's a solution and will happily upgrade to a paid plan if this is resolved.

MGit works great in this area, but you would then need to use a separate editor (like quick edit https://play.google.com/store/apps/details?id=com.rhmsoft.edit or https://billthefarmer.github.io/notes/) so working work flow but slightly less automated.

**Comment by vHanda** [comment]
Sorry about this issue guys. I just tried to reproduce it, and the files are getting deleted on my end.

https://github.com/GitJournal/foam/commit/a2df80a7a50d006e86e8664f35e4b13df0e4ffd5

I'm going to spend another 15-20 minutes now to try and reproduce this. If I can reproduce it, I'll happily fix it. I rarely delete notes since I mostly use GitJournal for journaling and not note taking, but I hate to see it so broken for all of you.

**Comment by davidluhr** [comment]
@vHanda Thank you so much! Excited to be able to use the app and again and I appreciate all your work to provide this for us!

**Comment by clach04** [comment]
> I've unfortunately had to uninstall the app to avoid the issue. I miss it; it was a valuable part of my personal knowledge management and without it I don't have a way to edit my notes without a computer.
> 
> I'm hoping there's a solution and will happily upgrade to a paid plan if this is resolved.

MGit works great in this area, but you would then need to use a separate editor (like quick edit https://play.google.com/store/apps/details?id=com.rhmsoft.edit or https://billthefarmer.github.io/notes/) so working work flow but slightly less automated.

**Comment by whirledsol** [comment]
> Nevermind. I managed to reproduce it. I understand what's going on. Let me try and fix it.

Any insights as to what could be causing this? Still occurs on Android, even after a fresh install.

The only thing that seems to work is to manually delete the root files both locally and externally and doing a sync.

Could this be related at all to `lib\repository.dart`:360
```dart
        if (ex is! GitRefNotFound) {
```

**Comment by davidluhr** [comment]
@whirledsol Good timing! I was about to follow up on this yesterday myself.

I hope there's a good fix for this. I haven't been able to use this app since April and it's definitely put a dent in my personal knowledge management.

I appreciate the effort to solve this and would happily pay for a subscription to use this app again.

**Comment by tschager** [comment]
I would also be very interested in a fix for this bug and would also be glad to help

**Comment by vHanda** [comment]
The problem was with my faulty git merge implementation in dart-git. I'd moved from libgit2, since cross compiling it was such a PITA, to a combination of go-git and my own implementation in dart.

I've fixed the bug in dart-git, however, while upgrading the flutter + dart version, GitJournal is completely broken as there is a regression in dart's zlib decoder and now reading packfiles is broken. I've reported [it](https://github.com/dart-lang/sdk/issues/56481#issuecomment-2292492206) upstream, lets see how it goes.

Either -
1. It's fixed upstream in time for the next release (September), so I'll use that Flutter version.
2. It's not fixed upstream and I -
    2.a - Find a workaround locally
    2.b - Revert the upgrade to the version of Flutter and Dart.
    
(1) seems more likely since upstream has already come up with a patch. So let's see.

--

Basically expect an update of GitJournal around September with this fix.

**Comment by clach04** [comment]
Fix for https://github.com/dart-lang/sdk/issues/56526 appears to be included in https://github.com/dart-lang/sdk/releases/tag/3.5.2

**Comment by vHanda** [comment]
Yup. But I already released version 1.89 with this fix on Android. I downgraded the flutter version.

Next month I'll upgrade flutter. It provides no big change to GitJournal.

**Comment by bezark** [comment]
Hello! Just wondering if this issue been resolved because unfortunately I believe I'm still experiencing it. Namely I have a bunch of files I'm trying to delete on my computer and they never register as deleted in the app, so whenever it pushes it just recreates them.  I'm happy to provide more context if that helps debugging!

**Comment by clach04** [comment]
> Hello! Just wondering if this issue been resolved because unfortunately I believe I'm still experiencing it. Namely I have a bunch of files I'm trying to delete on my computer and they never register as deleted in the app, so whenever it pushes it just recreates them. I'm happy to provide more context if that helps debugging!

Worth providing the exact version number and where you got the apk from for clarity.

**Comment by latepaul** [comment]
Hi I'm experiencing this bug with  1.89.0+3798, installed from Google Play yesterday.

**Comment by latepaul** [comment]
So I tried deleting the notes in the app and it failed.

Here's the sequence of events as far as I recall
- I deleted a note I'd created in the app - this worked and pushed the change to my git repo
- I then deleted several notes together, which had already been deleted by cli git client on my PC and pushed to the repo
- the notes appear to be deleted but if I then sync manually they re-appear. They are not in the repo though.
At this point I figured that the push from GitJournal is failing because the files are already gone at the repo end. So
- I add a new note and allow it to sync. This adds the new file and re-adds all the deleted files (as per this original bug)
- now I delete the notes in the app and this gets pushed to the repo.

So now that I sort of understand what's happening I can work-around it. 

TLDR: work-around is always delete from the app.

**Comment by latepaul** [comment]
BTW - this is an annoyance but overall this is a great app!

**Comment by tschager** [comment]
Is there anything we can do to help regarding this issue? I still have it in version 1.89.0+3789; this is such a great app, but not being able to delete notes from a computer is really inconvenient.

**Comment by clach04** [comment]
@tschager I gave up. I'm using [Markor ](https://github.com/gsantner/markor)and an Android git client:

  * https://github.com/maks/MGit
  * https://github.com/catpuppyapp/PuppyGit

**Comment by Aodhagan** [comment]
Without reliable version control this app is unusable. The maintainer is taking money via the paid pro version of the app (which I purchased out of support for OSS). Either maintain the app by fixing app breaking bugs or don't bother fix them and stop taking money for a "pro" version. Either one or the other. Regardless it's a definite uninstall for me of what would be an otherwise very useful app!

**Comment by whirledsol** [comment]
Thank you to @vHanda and everyone in this thread. GitJournal is a nice app but due to this bug and lack of progress in the app, I switched over to "Zettel Notes" which does a pretty decent job with folders and customization. Always willing to return and help more once things get going again.
